ast/TreeEquality.cc

```diff
@@ -302,39 +317,56 @@ bool structurallyEqual(const void *avoid, const Tag tag, const void *tree, const
     }
 }
 
-bool structurallyEqual(const void *avoid, const ExpressionPtr &tree, const ExpressionPtr &other, bool root) {
-    ENFORCE(tree != nullptr);
-    ENFORCE(tree != nullptr);
+bool structurallyEqual(const core::GlobalState &gs, const void *avoid, const ExpressionPtr &tree,
+                       const ExpressionPtr &other, const std::shared_ptr<spdlog::logger> logger,
+                       const core::Loc selectionLoc, bool root) {
+    ENFORCE(tree);
+    ENFORCE(other);
+    if (!tree) {
+        logger->error("msg=\"ExtractToVariable: tree is null in structurallyEqual\" selectionLoc=\"{}\"",
+                      selectionLoc.showRaw(gs));
+    }
+    if (!other) {
+        logger->error("msg=\"ExtractToVariable: other is null in structurallyEqual\" selectionLoc=\"{}\"",
+                      selectionLoc.showRaw(gs));
+    }
+    if (!tree || !other) {
+        logger->error("source=\"{}\"", absl::CEscape(selectionLoc.file().data(gs).source()));
+    }
```

Comment:
Do you want to `return false` here?

That way at least the user won't have Sorbet crash on account of our mistakes.